### PR TITLE
Record soft failure if docker-compose fails with dependency problems

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/93333

### CODE REVIEW

**Comment by jlausuch** [comment]
VR: https://openqa.suse.de/tests/6151651#

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by jlausuch** [comment]
VR in a test where it was working: https://openqa.suse.de/tests/6151664#step/docker_compose/50

**Comment by jlausuch** [comment]
I will also add a soft-failure for this bug: https://bugzilla.suse.com/show_bug.cgi?id=1186691

**Comment by jlausuch** [comment]
VR for the second check: https://openqa.suse.de/tests/6151803#step/docker_compose/53

File: tests/containers/docker_compose.pm [hunk 1]
```diff
@@ -42,8 +42,14 @@ sub run {
     add_suseconnect_product(get_addon_fullname('python2')) if is_sle('=15-sp1');
 
     record_info 'Test #1', 'Test: Installation';
-    zypper_call("in docker-compose");
-    assert_script_run 'docker-compose --version';
+    if (script_output('zypper -n in docker-compose', proceed_on_failure => 1) =~ /nothing provides python3-docker/) {
```
Comment (pdostal) [inline tests/containers/docker_compose.pm:45]:
```suggestion
    if (script_output('zypper -n se python3-docker', proceed_on_failure => 1) != 0) {
```

**Review by pdostal — CHANGES_REQUESTED** [review]
As we know we're trying to catch that `python3-docker` is missing we can just `zypper search` for it. Trying to install `docker-compose` using `script_output('zypper -n in` is in my opinion too aggressive.

**Comment by jlausuch** [comment]
VRs with latest change (not finished yet):

- https://openqa.suse.de/tests/6152780 (python check)
- https://openqa.suse.de/tests/6152782 (version check)
- https://openqa.suse.de/tests/6152781 (no bugs)

File: tests/containers/docker_compose.pm (hunk 1 above)
Comment (mloviska) [reply tests/containers/docker_compose.pm]: Maybe we should still consider using `zypper_call`.

Comment (jlausuch) [reply tests/containers/docker_compose.pm]: > Maybe we should still consider using `zypper_call`.

done

File: tests/containers/docker_compose.pm [hunk 2]
```diff
@@ -42,8 +42,15 @@ sub run {
     add_suseconnect_product(get_addon_fullname('python2')) if is_sle('=15-sp1');
 
     record_info 'Test #1', 'Test: Installation';
+    if (zypper_call('se python3-docker') != 0) {
+        record_soft_failure "bsc#1182675";
```
Comment (pdostal) [inline tests/containers/docker_compose.pm:46]:
```suggestion
        record_soft_failure "bsc#1182675 - python3-docker is missing";
```

File: tests/containers/docker_compose.pm [hunk 3]
```diff
@@ -42,8 +42,15 @@ sub run {
...
+        return 0;
+    }
     zypper_call("in docker-compose");
-    assert_script_run 'docker-compose --version';
+    if (script_output('docker-compose --version', proceed_on_failure => 1) =~ /distribution was not found/) {
+        record_soft_failure "bsc#1186691";
```
Comment (pdostal) [inline tests/containers/docker_compose.pm:51]:
```suggestion
        record_soft_failure "bsc#1186691 - docker-compose probably missing dependency";
```

**Review by pdostal — APPROVED** [review]
Just readability improvements. Otherwise I'm happy to merge when verification runs succeed :rocket:

**Comment by jlausuch** [comment]
> Just readability improvements. Otherwise I'm happy to merge when verification runs succeed 🚀

Done

**Review by grisu48 — APPROVED** [review]
LGTM :guitar:

**Comment by jlausuch** [comment]
[docker-compose --version check](https://openqa.suse.de/tests/6154320)
[python3-docker check](https://openqa.suse.de/tests/6154321)

**Comment by jlausuch** [comment]
It looks like the python3-docker dependency [has been solved](https://openqa.suse.de/tests/6156367#step/docker_compose/39) in the meantime. So, I am keeping only the soft failure for the bug about `--version`.

**Comment by jlausuch** [comment]
VRs are ok: 
docker-compose --version check: [12-SP4](https://openqa.suse.de/tests/6156368#step/docker_compose/36)  [12-SP5](https://openqa.suse.de/tests/6156369#step/docker_compose/48)
python3-docker check: [15-SP3](https://openqa.suse.de/tests/6156367#step/docker_compose/39)

**Pull request merged**